sql/sql_acl.cc

```diff
@@ -14312,6 +14313,50 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else
+  if (redirect_enabled)
+  {
+    int total_size = 18
+                   + strlen(redirect_server_host)
+                   + 1 + strlen(redirect_server_port)
+                   + 6 + strlen(sctx->user)
+                   + (!strcmp(redirect_server_ttl, "0") ? 0 : (5 + strlen(redirect_server_ttl)));;
+
+    if (total_size >= MAX_REDIRECTION_LEN)
+    {
+      sql_print_error("error", ("redirection info is too large to return to client (len= %lu)",
+                     total_size));
+      my_ok(thd);
+    }
+    else
+    {
+      char *msg = new char[MAX_REDIRECTION_LEN];
+      if (NULL == msg)
+      {
+        DBUG_RETURN(1);
+      }
+
+      msg[0] = '\0';
+      strcat(msg, "Location: mysql://");
+      strcat(msg, redirect_server_host);
```

Comment:
Will this support IPv6 addresses? If it will, using the [first one listed here](https://tools.ietf.org/html/rfc5952#section-6) might be a good idea as otherwise the address and the port are somwhat ambiguous.

Reply:
Great point, making the change.